src/main/java/org/damiane/service/impl/UserServiceImpl.java

```diff
@@ -1,58 +1,83 @@
-package org.damiane.service;
+package org.damiane.service.impl;
 
+import lombok.extern.slf4j.Slf4j;
 import org.damiane.entity.User;
 import org.damiane.repository.UserRepository;
+import org.damiane.service.AuthenticateService;
+import org.damiane.service.UserService;
 import org.damiane.util.PasswordGenerator;
 import org.damiane.util.UsernameGenerator;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
 
+@Slf4j
 @Service
 public class UserServiceImpl implements UserService {
 
     @Autowired
     private UserRepository userRepository;
 
+    @Autowired
+    private AuthenticateService authenticateService;
+
     @Override
-    public List<User> getAllUsers() {
+    public List<User> getAllUsers(String username, String password) {
+
+        authenticateService.matchUserCredentials(username, password);
+        log.info("User Authenticated Successfully");
+
         return userRepository.findAll();
     }
 
     @Override
-    public User getUserById(Long id) {
+    public User getUserById(Long id, String username, String password) {
+
+        authenticateService.matchUserCredentials(username, password);
+        log.info("User Authenticated Successfully");
+
         return userRepository.findById(id).orElse(null);
     }
 
     @Override
```

Comment:
Mind using  @Transactional only where it's necessary, same for classes in this package

Reply:
Removed Transactional annotation in places where it wasn't needed